src/cljs/bluegenes/route.cljs

```diff
@@ -0,0 +1,230 @@
+(ns bluegenes.route
+  (:require [re-frame.core :refer [subscribe dispatch reg-sub reg-event-db reg-event-fx reg-fx]]
+            [oops.core :refer [ocall]]
+            [reitit.core :as r]
+            [reitit.coercion :as rc]
+            [reitit.coercion.spec :as rcs]
+            [reitit.frontend :as rf]
+            [reitit.frontend.controllers :as rfc]
+            [reitit.frontend.easy :as rfe]))
+
+;; # Quickstart guide:
+;; (aka. I just want to route something but don't want to read all this code!)
+;; - Don't forget to import the route namespace!
+;; ```
+;; (require '[bluegenes.route :as route])
+;; ```
+;; - Href
+;; ```
+;; [:a {:href (route/href ::route/upload-step {:step "save"})} "My anchor"]
+;; ```
+;; - Dispatch (for on-click and event handlers)
+;; ```
+;; (dispatch [::route/navigate ::route/upload-step {:step "save"}])
+;; ```
+;; ## Order of arguments: `route params query`
+;; Only route is required, the rest are optional. Although you will get a
+;; warning if you use a route that expects params, without specifying params.
+;; Note that the `:mine` param is an exception to this, since it's the parent
+;; of all routes, so it gets injected automatically from db if not specified.
+
+;; Based on the official reitit frontend-re-frame example: (2019.06.18)
+;; https://github.com/metosin/reitit/tree/master/examples/frontend-re-frame
+
+;;; Events ;;;
+
+(reg-event-fx
+ ::navigate
+ (fn [{db :db} [_ route & [params query]]]
+   {::navigate! {:k route
+                 :query query
+                 :params (update params :mine #(or % (:current-mine db)))}}))
+
+(reg-event-db
+ ::navigated
```

Comment:
could we add a comment about when/why this is used?

Reply:
Is it worth it? It's pretty internal to how the router works, so it's not that important to know how it works, and those who do want to know can read it as it's still a pretty short and simple function. [=
I at least can't think of a succinct and useful comment to put there.

Reply:
>"you probably won't need to use this as it's an internal helper method" ?
Sure, marking it as internal sounds reasonable.